app/src/main/java/com/aaronjwood/portauthority/network/Wireless.java

```diff
@@ -186,6 +186,38 @@ public int getInternalWifiSubnet() {
         return netmask;
     }
 
+
+    public int getInternalWifiIpAddressAsInt() {
+        int ip = this.getWifiInfo().getIpAddress();
+        if (ByteOrder.nativeOrder().equals(ByteOrder.LITTLE_ENDIAN)) {
+            ip = Integer.reverseBytes(ip);
+        }
+
+        byte[] ipByteArray = BigInteger.valueOf(ip).toByteArray();
+
+        try {
+            return new BigInteger(InetAddress.getByAddress(ipByteArray).getAddress()).intValue();
+        } catch (UnknownHostException ex) {
+            return 0;
+        }
+    }
+
+
+    /**
+     * Returns the number of hosts in the subnet.
+     *
+     * @return Number of hosts as an integer.
+     */
+    public int getNumberOfHostsInWifiSubnet(){
```

Comment:
You could make this static and reuse it in `ScanHostsAsyncTask` in `doInBackground` since you're getting the number of hosts there too.